modules/internal/ArrayViewRankChange.chpl

```diff
@@ -22,9 +22,337 @@
+        if !collapsedDim(d) {
+          downDim += 1;
+          if (upDim == downDim) then
+            return downdom.dsiDim(d);
+        }
+      }
+      halt("Fell out of loop in dsiDim():", (upDim, downDim));
+    }
+
+    proc dsiDims() {
+      var upDims: rank*range(idxType, stridable=stridable);
+      var upDim = 1;
+      for param downDim in 1..downrank {
+        if !collapsedDim(downDim) {
+          upDims(upDim) = downdom.dsiDim(downDim);
+          upDim += 1;
+        }
+      }
+      return upDims;
+    }
+
+    proc dsiGetIndices() {
+      return updom.dsiGetIndices();
+    }
+
+    proc dsiSetIndices(inds) {
+      pragma "no auto destroy"
+      var updomRec = {(...inds)};
+      updom = updomRec._value;
+      var downdomclass = dist.downdist.newRectangularDom(rank=downrank,
+                                                         idxType=idxType,
+                                                         stridable=stridable);
+      pragma "no auto destroy"
+      var downdomLoc = _newDomain(downdomclass);
+      downdomLoc = chpl_rankChangeConvertDom(inds, inds.size, collapsedDim, idx);
+      downdomLoc._value._free_when_no_arrs = true;
+      downdomPid = downdomLoc._pid;
+      downdomInst = downdomLoc._instance;
+    }
+
+    proc dsiMember(i) {
+      return downdom.dsiMember(chpl_rankChangeConvertIdx(i, collapsedDim, idx));
+    }
+
+    iter these() {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom do
+          yield i;
+      } else {
+        for i in downdom do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.standalone {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag) do
+          yield i;
+      } else {
+        for i in downdom.these(tag) do
+          yield downIdxToUpIdx(i);
+      }
+    }
+
+    iter these(param tag: iterKind) where tag == iterKind.leader {
+      if chpl__isDROrDRView(downdom) {
+        for followThis in updom.these(tag) do
+          yield followThis;
+      } else {
+        for followThis in downdom.these(tag) {
+          const followThisLoD = chpl_rankChangeConvertHiDTupleToLoD(followThis);
+          yield followThisLoD;
+        }
+      }
+    }
+
+    iter these(param tag: iterKind, followThis)
+      where tag == iterKind.follower {
+      if chpl__isDROrDRView(downdom) {
+        for i in updom.these(tag, followThis) do
+          yield i;
+      } else {
+        const followThisHiD = chpl_rankChangeConvertLoDTupleToHiD(followThis);
+        for i in downdom.these(tag, followThisHiD) {
+          yield chpl_rankChangeConvertIdxHiDToLoD(i, collapsedDim, idx, rank);
+        }
+      }
+    }
+
+    proc chpl_rankChangeConvertLoDTupleToHiD(tup) {
+      var tupHiD: downrank*tup(1).type;
+      var i = 1;
+      for param d in 1..downrank do
+        if collapsedDim(d) then
+          tupHiD(d) = 0..0;  // TODO: This seems bogus/arbitrary... or is it?
```

Comment:
I've been meaning to write a test to see if this is buggy or not.  I'll add it to the TODO list.  I was guessing it needed to be something like that too, put this in as a placeholder in my first draft of the code for reasons I don't remember and was surprised it never failed.  I haven't thought through it enough to determine whether that means we're not testing this routine (likely... I'm working on a userAPI test like the one I wrote for arrays to make sure we exercise all routines in a simple way) or whether the fact that the dimension is getting compressed out anyway means that the index doesn't matter.